playwright-e2e/dsm/component/tablist.ts

```diff
@@ -0,0 +1,71 @@
+import { Locator, Page, expect } from '@playwright/test';
+import { Tab } from 'dsm/enums';
+import ContactInformationTab from 'dsm/pages/tablist/contact-information-tab';
+import GenomeStudyTab from 'dsm/pages/tablist/genome-study-tab';
+import OncHistoryTab from 'dsm/pages/tablist/onc-history-tab';
+import SampleInformationTab from 'dsm/pages/tablist/sample-information-tab';
+import SurveyDataTab from 'dsm/pages/tablist/survey-data-tab';
+import SequencingOrderTab from 'dsm/pages/tablist/sequencing-order-tab';
+import MedicalRecordsTab from 'dsm/pages/tablist/medical-records-tab';
+import { waitForNoSpinner } from 'utils/test-utils';
+
+export default class Tablist {
+  protected readonly tabs = new Map<string, object>([
+    [Tab.CONTACT_INFORMATION, new ContactInformationTab(this.page)],
+    [Tab.SAMPLE_INFORMATION, new SampleInformationTab(this.page)],
+    [Tab.GENOME_STUDY, new GenomeStudyTab(this.page)],
+    [Tab.ONC_HISTORY, new OncHistoryTab(this.page)],
+    [Tab.MEDICAL_RECORD, new MedicalRecordsTab(this.page)],
+    [Tab.SURVEY_DATA, new SurveyDataTab(this.page)],
+    [Tab.SEQUENCING_ORDER, new SequencingOrderTab(this.page)],
+  ]);
+
+  public constructor(private readonly page: Page, private readonly tab: Tab | string) {}
+
+  get toLocator(): Locator {
+    return this.page.locator(`xpath=//tabset//*[@role="tablist"]//a[@role="tab"][.//text()[normalize-space()="${this.tab}"]]`);
+  }
+
+  public async isVisible(): Promise<boolean> {
+    try {
+      await expect(this.toLocator).toBeVisible();
+    } catch (err) {
+      return false;
+    }
+
+    let contactInformationEntered = true;
+    if (this.tab === Tab.CONTACT_INFORMATION) {
+      await this.checkAndClick();
+      contactInformationEntered = await this.HasContactInformationData();
+    }
+    return contactInformationEntered;
+  }
+
+  protected async isActive(): Promise<boolean> {
+    const clas = await this.toLocator.getAttribute('class');
+    return clas ? clas.includes('active') : false;
+  }
+
+  public async click<T extends object>(): Promise<T> {
+    await expect(this.toLocator, `Tab "${this.tab}" is not enabled`).toBeEnabled();
+    await this.checkAndClick();
+    return (this.tabs as Map<string, object>).get(this.tab) as T;
+  }
+
+  protected async HasContactInformationData(): Promise<boolean> {
```

Comment:
small thing but h in has should probably be lower case

Reply:
Good catch 👍🏻